rlclientlib/live_model_impl.cc

```diff
@@ -397,6 +397,13 @@ namespace reinforcement_learning {
     _learning_mode = learning::to_learning_mode(_configuration.get(name::LEARNING_MODE, value::LEARNING_MODE_ONLINE));
   }
 
+  live_model_impl::~live_model_impl() {
+    if (_interaction_logger)
```

Comment:
how flushing in live_model destructor is better than doing it in async_batcher's one?

Reply:
Because of liveness. 
live_model dtor owns all the relevant objects and they will be valid during its dtor.
async_batcher doesn't own anything, so it's all potentially dangling pointers when its dtor is called.

I explained it in detail over one of the commits.

```
Add a dtor to live_model_impl and flush the queues from it.

Why flushing from live_model_impl is ok but from async_batcher is not?

async_batcher holds un-owned references to a lot of stuff and the
expectation is that those objects won't be valid during destruction.

Flushing data in async_batcher requires those dangling pointers to
be valid so we cannot do it from the dtor.

Then why is it ok to do it from live_model_impl dtor? As it turns out,
it holds owning references to pretty much everything that's needed for flushing.

Did you say pretty much? Yes, there are a lot of un-owned references in
live_model_impl as well but they look related to initialization only.
Life is too short otherwise.
```

Reply:
yeah that makes sense, thank you